src/go/rpk/pkg/cli/cmd/redpanda/config.go

```diff
@@ -104,6 +105,7 @@ func bootstrap(fs afero.Fs) *cobra.Command {
 			p := config.ParamsFromCommand(cmd)
 			cfg, err := p.Load(fs)
 			out.MaybeDie(err, "unable to load config: %v", err)
+			cfg = cfg.FileOrDefaults()
```

Comment:
Inline comment, `// we modify fields in the raw file without writing env / flag overrides`

Also, I think we need an issue tracking what the purpose of `bootstrap` and `init` are. I'd expect bootstrap to be everything start does _except_ executing redpanda, but that's not the case.